env/interpolator.go

```diff
@@ -2,156 +2,162 @@ package env
 
 import (
 	"fmt"
-	"os"
 	"regexp"
 	"strconv"
 	"strings"
 )
 
 var variablesWithBracketsRegex = regexp.MustCompile(`([\\\$]?\$\{([^}]+?)})`)
 var variablesWithNoBracketsRegex = regexp.MustCompile(`([\\\$]?\$[a-zA-Z0-9_]+)`)
-
 var substringRegexp = regexp.MustCompile(`\A\s*:\s*(\-?\s*\d+)(?:\s*:\s*(\-?\s*\d+))?\s*\z`)
 
-func Interpolate(str string) (interpolated string, err error) {
+// Interpolate takes a string and interpolates shell-like variable expansions in the provided string
+func (e Environment) Interpolate(str string) (string, error) {
```

Comment:
This is functionally the same, I just used the go pattern of "fail fast" to reduce nesting depth.